Negated condition

colinmcardell opened the pull request.

colinmcardell committed:
Merge remote-tracking branch 'Kabal/master'

colinmcardell committed:
Implements `OPNegatedCondition` with appropriate documentation.

colinmcardell committed:
Moved negated condition error key into `NSError+Operation`

Attempting to follow suit with the other error keys, collected in one
location within the framework.

pronebird commented on Pod/Classes/Conditions/OPNegatedCondition.m:
```diff
@@ -0,0 +1,116 @@
+
+- (instancetype)init
+{
+    @throw [NSException exceptionWithName:[NSString stringWithFormat:@"Method Unavailable"]
```
I think it would be better to give a hint like:

``` objc
- (instancetype)init {
    [NSException raise:NSGenericException format:@"Please use -initWithXXX: instead."];

    return nil;
}
```

colinmcardell committed:
Making exception that is thrown by unavailable `-init` within `OPNegatedCondition` and `OPSilentCondition` more explicit.
- Using `NSInternalInconsistencyException` (not `NSGenericException` because that would be more for an unknown exception).
- returning `nil` within `-init`
All really moot because the `-init` method is `NS_UNAVAILABLE` which causes the compiler to get angry if you try to use it, and `-init` is implemented just to make the compiler happy in the first place due to `NS_DESIGNATED_INITIALIZER`
